### PR TITLE
docs: add goals and intro for conformane profiles GEP

### PR DESCRIPTION
**Pull request by shaneutt**
**What type of PR is this?**

/kind gep

**What this PR does / why we need it**:

This is one of several issues needed in support of #1709.

This is meant to be a small initial iteration to make sure we have alignment on the "what" and "why" by way of the tldr, goals and introduction, but not bothering yet with the how. The hope is that if we have alignment follow up iterations can be responsible for filling in the "how". Please feel very free to add suggestions for anything you want to see changed or added, I want to incorporate and co-author this with anyone with ideas and interest.

### CODE REVIEW

**Comment by youngnick** [comment]
This LGTM as starting point, thanks @shaneutt!

**Review by Xunzhuo — APPROVED** [review]
LGTM, thanks @shaneutt!

**Comment by LiorLieberman** [comment]
I like the idea of splitting it to iterations, thanks @shaneutt !
/lgtm
/approve

**Comment by shaneutt** [comment]
Forgot to put a hold on this to allow @robscott and/or @youngnick to give it a look. Please do one or both of you check it out, but given that it's `provisional` I think it's reasonable to make small PRs for any changes or adjustments desired now, rather than reverting?

**Comment by youngnick** [comment]
That sounds fine to me.

File: site-src/geps/gep-1709.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# GEP-1709: Conformance Profiles
```
Comment (robscott) [inline site-src/geps/gep-1709.md:1]: Needs a link in mkdocs.yml since we haven't automated this yet.

**Review by robscott — COMMENTED** [review]
Thanks @shaneutt! LGTM other than one small nit that can be a follow up.

File: site-src/geps/gep-1709.md (hunk 1 above)
Comment (shaneutt) [reply site-src/geps/gep-1709.md:1]: Will follow up, thank you :+1:

**Pull request merged**